Ok, so I did a quick test and I may have found a related bug, Path object not having a `.type`.

```
Traceback (most recent call last):
  File "/Users/Documents/pyv/py3/lib/python3.7/site-packages/visidata/threads.py", line 201, in _toplevelTryFunc
    t.status = func(*args, **kwargs)
  File "/Users/.visidata/save_excel.py", line 50, in save_xls
    if not isNumeric(v):
  File "/Users/Documents/pyv/py3/lib/python3.7/site-packages/visidata/_types.py", line 72, in isNumeric
    return col.type in (int,vlen,float,currency,date)
  File "/Users/Documents/pyv/py3/lib/python3.7/site-packages/visidata/path.py", line 78, in __getattr__
    r = getattr(self._path, k)
AttributeError: 'PosixPath' object has no attribute 'type'
```

Without a fix to give `Path` a `.type` the following works, but I would assume that you'll want the Path.type fixed instead of hacking it 🙂:

```
try: v = v if isNumeric(v) else str(v)
except: v = str(v)
```

If this is fixed, we can use isNumeric.

--- a/visidata/loaders/xlsx.py
+++ b/visidata/loaders/xlsx.py
@@ -70,12 +70,13 @@ def save_xlsx(vd, p, *sheets):
         for dispvals in vs.iterdispvals(format=False):
 
             row = []
-            for v in dispvals.values():
-                if type(v) == date:
+            for col, v in dispvals.items():
+                if col.type == date:
                     v = datetime.datetime.fromtimestamp(int(v.timestamp()))
-                elif not type(v) in [int,float,str]:
+                elif not isNumeric(col):
                     v = str(v)
                 row.append(v)
+
             ws.append(row)
 
     wb.active = ws
@@ -97,13 +98,9 @@ def save_xls(vd, p, *sheets):
         for col_i, col in enumerate(vs.visibleCols):
             ws1.write(0, col_i, col.name)
 
-        for r_i, dispvals in enumerate(vs.iterdispvals(format=False)):
+        for r_i, dispvals in enumerate(vs.iterdispvals(format=True)):
             r_i += 1
             for c_i, v in enumerate(dispvals.values()):
-
-                if not type(v) in [int,float,str]:
-                    v = str(v)
-
                 ws1.write(r_i, c_i, v)
 
     wb.save(p)

